src/buildstream/_artifactcache.py

```diff
@@ -410,7 +414,9 @@ def _push_artifact_blobs(self, artifact, artifact_digest, remote):
         except grpc.RpcError as e:
             if e.code() != grpc.StatusCode.RESOURCE_EXHAUSTED:
                 raise ArtifactError(
-                    "Failed to push artifact blobs with status {}: {}".format(e.code().name, e.details())
+                    "Failed to push artifact blobs with status {}: {}".format(
+                        e.code().name, e.details(), temporary=True
```

Comment:
mypy test is failing because temporary is used as parameter to format rather than ArtifactError